flutter_local_notifications/example/lib/main.dart

```diff
@@ -207,6 +205,21 @@ class _HomePageState extends State<HomePage> {
         );
   }
 
+  Future<void> _requestAndroidPermissions() async {
+    final AndroidFlutterLocalNotificationsPlugin? androidImplementation =
+        flutterLocalNotificationsPlugin.resolvePlatformSpecificImplementation<
+            AndroidFlutterLocalNotificationsPlugin>();
+
+    final bool? granted = await androidImplementation?.requestPermission();
+
+    await showDialog<void>(
```

Comment:
If the intent behind having a separate button to request permissions was done to display the result, if the example app can be updated to do this on startup via `initState` then an extra label could be added to the example to display the result. By that I mean like a `bool` field added and `setState()` can be called to save the result of the call and update the label that indicates if permissions were granted on Android